conmon/conmon.c

```diff
@@ -1375,11 +1380,38 @@ int main(int argc, char *argv[])
 			 "--pid-file", opt_pid_file,
 			 NULL);
         } else {
+		char *command;
+		if (opt_restore_path)
+			command = "restore";
+		else
+			command = "create";
+
 		add_argv(runtime_argv,
-			 "create",
+			 command,
 			 "--bundle", opt_bundle_path,
 			 "--pid-file", opt_pid_file,
 			 NULL);
+
+		if (opt_restore_path) {
+			/*
+			 * 'runc restore' is different from 'runc create'
+			 * as the container is immediately running after
+			 * a restore. Therefore the '--detach is needed'
+			 * so that runc returns once the container is running.
+			 *
+			 * '--image-path' is the path to the checkpoint
+			 * which will be become important when using pre-copy
+			 * migration where multiple checkpoints can be created
+			 * to reduce the container downtime during migration.
+			 *
+			 * '--work-path' is the directory CRIU will run in and
+			 * also place its log files.
+			 */
+			add_argv(runtime_argv, "--detach",
```

Comment:
Will adding `--detach` prevent us from attaching to the container once it has been restored?

Reply:
No, those are different attaches. The podman attach is using the attach socket to talk to the container and that already works. I already tried it with registry.fedoraproject.org/f26/httpd and after a restore I can see the request to the httpd server being logged on ```podman attach -l```

The ```runc restore --detach``` is that same detach as ```runc run --detach``` which will immediately return to the shell.

Reply:
Ack, just wanted to make sure